deepmd/pt/entrypoints/main.py

```diff
@@ -105,8 +105,18 @@ def get_trainer(
     local_rank = os.environ.get("LOCAL_RANK")
     if local_rank is not None:
         local_rank = int(local_rank)
-        assert dist.is_nccl_available()
-        dist.init_process_group(backend="nccl")
+        nccl_available = dist.is_nccl_available()
+        gloo_available = dist.is_gloo_available()
+        # nccl first
+        if nccl_available:
+            backend = "nccl"
+        elif gloo_available:
+            backend = "gloo"
```

Comment:
I have a question: when one installs the GPU version but doesn't have a GPU (or set CUDA_VISIBLE_DEVICES to empty), will backend be gloo?